### DISCUSSION TITLE
Support for omitting `@updatedAt `?

### DISCUSSION THREAD

**Discussion by crishoj** [m1212e/prismabox#4] [Q&A] [answered]
First of all, thanks for this very useful fork of [prisma-typebox-generator](https://github.com/adeyahya/prisma-typebox-generator).

I was wondering if there is support for omitting `updatedAt` from the data models, similar to how `createdAt` is omitted:

> `createdAt` will be detected and ignored if it follows exactly this pattern: `createdAt DateTime @default(now())`

Example schema:

```prisma
model Item {
  id             Int      @id @default(autoincrement())
  createdAt      DateTime @default(now())
  updatedAt      DateTime @updatedAt
  name           String   @unique
}
```

For this schema, `updatedAt` is filled/updated by Prisma itself, and would not be required in the input data.

**Comment by m1212e** [comment]
Hey, sorry for the delayed response: In the latest version this will be the case for input models. Does this solve your problem?

**Reply by crishoj** [reply]
This sounds like exactly what I need, but with https://github.com/m1212e/prismabox/releases/tag/1.1.0 I can't see that fields like `createdAt` and `updatedAt` are optional in the generated `Plain` type:

```ts
    createdAt: t.Date({ additionalProperties: false }),
    updatedAt: t.Date({ additionalProperties: false }),
```

Schema:

```prisma
  createdAt      DateTime       @default(now())
  updatedAt      DateTime       @updatedAt
```

Previously, I used the generated `Data` type with added `/// @prismabox.hide.data` declarations to omit `updatedAt`.

**Reply by m1212e** [reply, marked as answer]
With your above example I get

```ts
export const ItemPlain = Type.Object(
  {
    id: Type.Integer({ additionalProperties: false }),
    createdAt: Type.Date({ additionalProperties: false }),
    updatedAt: Type.Date({ additionalProperties: false }),
    name: Type.String({ additionalProperties: false }),
  },
  { additionalProperties: false },
);
```
As plain model and
```ts
export const ItemPlainInput = Type.Object(
  { name: Type.String({ additionalProperties: false }) },
  { additionalProperties: false },
);
```
as input model. Please note the rephrase of data models to input models. You might need to check the updated readme and adjust your settings to make the new version work as expected.

**Reply by crishoj** [reply]
`inputModel = true` did the trick, thanks!
